extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -52,9 +52,13 @@ export class AmpExperiment extends AMP.BaseElement {
 
     /** @private @const {!Promise<Object<string, ?string>>} */
     this.experimentVariants_ = Promise.all(variants).then(() => results);
-    this.experimentVariants_.then(this.addToBody_.bind(this));
+    const bodyMutatePromise =
+        this.experimentVariants_.then(this.addToBody_.bind(this));
 
     getService(this.getWin(), 'variant', () => this.experimentVariants_);
+    // Body mutation makes amp-experiment a render-delaying-extension.
+    // Use the service framework to inform outside when the mutation is done.
+    getService(this.getWin(), 'amp-experiment', () => bodyMutatePromise);
```

Comment:
This service `amp-experiment` will conflict with the implicitly registered service created with `AMP.registerElement`. With that, this callback will be discarded and any calls to `getService('amp-experiment')` will return the blank object from register.

I think the best approach here is to map from a render delaying extension to the service promise that must resolve before render is unblocked. Then we can name this service whatever we want.